pkg/kubelet/status/generate.go

```diff
@@ -29,8 +29,102 @@ const (
 	PodCompleted             = "PodCompleted"
 	ContainersNotReady       = "ContainersNotReady"
 	ContainersNotInitialized = "ContainersNotInitialized"
+	ReadinessGatesNotReady   = "ReadinessGatesNotReady"
 )
 
+// GeneratePodReadyCondition returns ready condition if all containers in a pod are ready, else it
+// returns an unready condition.
+func GeneratePodReadyConditions(spec *v1.PodSpec, conditions []v1.PodCondition, containerStatuses []v1.ContainerStatus, podPhase v1.PodPhase) (podReady v1.PodCondition, containersReady v1.PodCondition) {
+	setReadyConditionsFunc := func(condition v1.PodCondition) {
```

Comment:
Would prefer if this returned the 2 values and then assigned them at the call-site.  This took me too long to figure out..

Reply:
hmmm. It does not return in all cases though.